src/EnergyPlus/VariableSpeedCoils.cc

```diff
@@ -6565,14 +5860,16 @@ namespace VariableSpeedCoils {
 
         // note: load side is the evaporator side, and source side is the condenser side
 
-        int CondInletNode = state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).WaterInletNodeNum;
-        int CondOutletNode = state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).WaterOutletNodeNum;
+        auto &varSpeedCoil = state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum);
+        int CondInletNode = varSpeedCoil.WaterInletNodeNum;
+        int CondOutletNode = varSpeedCoil.WaterOutletNodeNum;
         // If heat pump water heater is OFF, set outlet to inlet and RETURN
-        if (PartLoadRatio == 0.0) {
+        if (PartLoadRatio == 0.0 || varSpeedCoil.availSched->getCurrentVal() <= 0.0) {
             state.dataLoopNodes->Node(CondOutletNode) = state.dataLoopNodes->Node(CondInletNode);
+            varSpeedCoil.SimFlag = false;
             return;
         } else {
```

Comment:
If the coil is scheduled off, set the `SimFlag` to `false` and return before doing the calc.